src/libPMacc/include/dataManagement/DataConnector.hpp

```diff
@@ -216,13 +238,7 @@ namespace PMacc
         {
             log< ggLog::MEMORY >( "DataConnector: being destroyed (%1% datasets left to destroy)" ) %
                                 datasets.size();
-            //datasets.clear();
-            for( auto it = datasets.begin(); it != datasets.end(); ++it )
-            {
-                log< ggLog::MEMORY >( "DataConnector: destroying %1%" ) %
-                                    (*it)->getUniqueId();
-                datasets.erase( it );
```

Comment:
that construct destroyed the data structure it looped over.
segfault if some objects forgot to `unshare` themselves